### ISSUE TITLE
`String#match` vs `String#scan`

### ISSUE THREAD

**Issue by AlexWayfer** [JuanitoFatas/fast-ruby#185]
Hello.

These methods are pretty similar.

I've done some benchmarks and figured out that `match` is twice faster than `scan`.

I can try to make a PR, but not sure do we need for it here.

**Comment by ParadoxV5** [comment]
Friendly reminder: `String#match` outputs a `MatchData` while `String#scan` outputs an `Array`.

**Comment by AlexWayfer** [comment]
> Friendly reminder: `String#match` outputs a `MatchData` while `String#scan` outputs an `Array`.

Yes, but they're similar, but an outside code will be different for the same input data (`String` and `Regexp`). But the point of comparison of these approaches is they complete the similar goal.

**Comment by mateusdeap** [comment]
@AlexWayfer If you're still interested in this, feel free to open a PR. All you need to do is follow the guidelines in the (CONTRIBUTING.md)[https://github.com/fastruby/fast-ruby/blob/main/CONTRIBUTING.md].
